src/NLog/Internal/TargetWithFilterChain.cs

```diff
@@ -152,12 +158,59 @@ static private bool GetTargetsByLevelForLogger(string name, List<LoggingRule> lo
             for (int i = 0; i <= LogLevel.MaxLevel.Ordinal; ++i)
             {
                 TargetWithFilterChain tfc = targetsByLevel[i];
-                tfc?.PrecalculateStackTraceUsage();
+                if (tfc is null)
+                    continue;
+
+                if (finalMinLevelWithFilters?.Count > 0)
+                {
+                    var finalMinLevelFilters = finalMinLevelWithFilters[i];
+                    if (finalMinLevelFilters.Value?.Count > 0)
+                    {
+                        targetsByLevel[i] = tfc = AppendFinalMinLevelFilters(tfc, finalMinLevelFilters.Value, finalMinLevelFilters.Key.Value);
+                    }
+                }
+
+                tfc.PrecalculateStackTraceUsage();
             }
 
             return targetsFound;
         }
 
+        private static bool LoggingRuleHasFinalMinLevelFilters(LoggingRule rule)
+        {
+            return LogLevel.Off != rule.FinalMinLevel && rule.Filters.Count != 0 && rule.Targets.Count == 0;
+        }
+
+        private static void CollectFinalMinLevelFiltersFromRule(LoggingRule rule, ref IList<KeyValuePair<FilterResult?, IList<Filter>>> finalMinLevelWithFilters)
+        {
+            finalMinLevelWithFilters = finalMinLevelWithFilters ?? new KeyValuePair<FilterResult?, IList<Filter>>[LogLevel.MaxLevel.Ordinal + 1];
+            for (int i = 0; i <= LogLevel.MaxLevel.Ordinal; ++i)
+            {
+                if (i < rule.FinalMinLevel.Ordinal)
```

Comment:
In other places in the codebase FinalMinLevel is either guarded for null or there are prevention to set it to null but it's default appears to be null. When testing 5.3.1 in our codebase (to update) we encountered a rule with a FinalMinLevel unset (null) that causes a null reference to be thrown here.

Reply:
Thank you for reporting this. I have been able to reproduce the NullReferenceException by creating a LoggingRule with filters but without targets. Will try get a fix out ASAP.

Reply:
@BooTeK NLog v5.3.2 has now been released - https://www.nuget.org/packages/NLog/5.3.2